boxoffice/models/line_item.py

```diff
@@ -235,36 +236,58 @@ def counts_per_date_per_item(item_collection, user_tz):
-    Accepts a list of dates.
-    ['2016-01-01', '2016-01-02'] => {'2016-01-01': }
+    Takes a timezone aware datetime object, item_ids and returns the sales amount accrued
+    during the day for the given datetime.
     """
     if not item_ids:
         return None
 
-    date_sales = {}
+    start_at = midnight_to_utc(sales_datetime, timezone=timezone)
+    end_at = midnight_to_utc(sales_datetime + datetime.timedelta(days=1), timezone=timezone)
+    sales_on_date = db.session.query('sum').from_statement('''SELECT SUM(final_amount) FROM line_item
+        WHERE status=:status AND ordered_at >= :start_at AND ordered_at < :end_at
+        AND line_item.item_id IN :item_ids
+        ''').params(status=LINE_ITEM_STATUS.CONFIRMED, start_at=start_at, end_at=end_at, item_ids=tuple(item_ids)).first()
+    return sales_on_date[0] if sales_on_date[0] else Decimal(0)
 
-    for sales_date in dates:
-        sales_on_date = db.session.query('sum').from_statement('''SELECT SUM(final_amount) FROM line_item
-            WHERE status=:status AND DATE_TRUNC('day', line_item.ordered_at AT TIME ZONE 'UTC' AT TIME ZONE :timezone)::date = :sales_date
-            AND line_item.item_id IN :item_ids
-            ''').params(timezone=user_tz, status=LINE_ITEM_STATUS.CONFIRMED, sales_date=sales_date, item_ids=tuple(item_ids)).first()
-        date_sales[sales_date] = sales_on_date[0] if sales_on_date[0] else Decimal(0)
-    return date_sales
 
-
-def sales_delta(user_tz, item_ids):
+def calculate_weekly_sales(item_collection_ids, user_tz, year):
     """
-    Calculates the percentage difference in sales between today and yesterday
+    Calculates sales per week for items in the given set of item_collection_ids in a given year,
+    in the user's timezone.
     """
-    today = datetime.datetime.now().strftime("%Y-%m-%d")
-    yesterday = (datetime.datetime.now() - datetime.timedelta(days=1)).strftime("%Y-%m-%d")
-    sales = sales_by_date([today, yesterday], user_tz, item_ids)
-    if not sales or not sales[yesterday]:
+    ordered_week_sales = OrderedDict()
+    for year_week in Week.weeks_of_year(year):
+        ordered_week_sales[year_week[1]] = 0
+    start_at = isoweek_datetime(year, 1)
+    end_at = isoweek_datetime(year+1, 1)
+
+    week_sales = db.session.query('sales_week', 'sum').from_statement(db.text('''SELECT EXTRACT(WEEK FROM ordered_at)
+        AS sales_week, SUM(final_amount)
+        FROM line_item INNER JOIN item on line_item.item_id = item.id
+        WHERE status=:status AND item_collection_id IN :item_collection_ids
+        AND ordered_at >= :start_at AND ordered_at < :end_at
+        GROUP BY sales_week ORDER BY sales_week;
+        ''')).params(timezone=user_tz, status=LINE_ITEM_STATUS.CONFIRMED,
+        start_at=start_at, end_at=end_at, item_collection_ids=tuple(item_collection_ids)).all()
+
+    for week_sale in week_sales:
+        ordered_week_sales[int(week_sale[0])] = week_sale[1]
```

Comment:
Is the `int()` call necessary? I'd have assumed the data would be integers already.